collections/ansible_collections/cloudkit/config_as_code/README.md

```diff
@@ -0,0 +1,225 @@
+# Ansible Collection - cloudkit.config_as_code
+
+This role defines the configuration as code of AAP to run the Ansible playbooks
+that are part of CloudKit.
+
+## Credentials
+
+Since AAP runs in an OpenShift cluster, we expect the required credentials to
+be passed as environment variables. These environment variables are defined as
+`Secrets` in OpenShift, and used by a container group.
+
+We define 2 container groups in AAP, one used to reconcile the configuration of
+AAP itself, and one used for cluster fulfillment operations.
+
+### Config-as-code environment variables
+
+In order to reconcile the configuration of AAP, we require the following
+variables:
+
+- `RH_USERNAME` and `RH_PASSWORD`: your Red Hat console username and password,
+  used to manage the subscription of AAP
+- `AAP_HOSTNAME`: URL of the AAP instance to be configured
+- `AAP_VALIDATE_CERTS`: true if the SSL certificate behind `AAP_HOSTNAME`
+  should be checked
+- `AAP_ORGANIZATION_NAME`: the AAP organization that should be created
+- `AAP_PROJECT_NAME`: the name of the project to be created
+- `AAP_PROJECT_GIT_URI`: the repository that is behind the AAP project
+- `AAP_PROJECT_GIT_BRANCH`: the git branch to use
+- `AAP_EE_IMAGE`: the registry URL to the execution environment image
+
+These variables must be defined in a secret named:
+`${AAP_ORGANIZATION_NAME}-${AAP_PROJECT_NAME}-config-as-code-ig` in the
+namespace where AAP is deployed.
+
+Note: since the container group is configured to run in the same namespace as
+the AAP instance, the admin credentials to log against AAP are injected into
+the pod.
+
+### Cluster fulfillment environment variables
+
+Here we need to define all the credentials required by the cluster fulfillment
+use case, e.g.:
+
+- `AWS_*`: AWS credentials
+- `OS_*`: OpenStack credentials
+- ...whatever in needed
+
+These variables must be defined in a secret named:
+`${AAP_ORGANIZATION_NAME}-${AAP_PROJECT_NAME}-cluster-fulfillment-ig` in the
+namespace where AAP is deployed.
+
+The cluster fulfillment needs to access the Kube API of the cluster it runs on,
+so we expect a service account `cloudkit-sa` to exists with enough rights (TBD).
+
+## Deploy a local AAP installation using CRC
+
+### Download and deploy CRC
+
+CRC basically OpenShift in a VM, you can download it from Red Hat Console:
+https://console.redhat.com/openshift/downloads. You will need `oc` which is
+downloadable from the same place.
+
+Extract the package, put the binary in your path, start the cluster, and log as
+`kubeadmin`:
+
+```
+crc start
+oc login -u kubeadmin  https://api.crc.testing:6443
+```
+
+### Deploy AAP
```

Comment:
There seems to be a step missing here; you've got (1) set up a CRC environment then (2) create an AnsibleAutomationPlatform CR, but where is the "install AAP" step? Or does CRC now include AAP out of the box?

Reply:
Right, I forgot to talk about the subscription, I will add it, thanks!

Reply:
Added!